### ISSUE TITLE
track moving car for each frame

### ISSUE THREAD

**Issue by raphaelschwinger** [raphaelschwinger/cone_reconstruction#16]
Output the position of the car at each timestamp

# Deliverables 
* a file with {position, time data}

**Comment by raphaelschwinger** [comment]
https://vuamitom.github.io/2019/12/13/fast-iterate-through-video-frames.html

**Comment by rxxxxxxb** [comment]
# Tracking the car :

## Methods that we can use: 

- CSRT Tracker([OpenCV Library](https://learnopencv.com/object-tracking-using-opencv-cpp-python/)): Discriminative Correlation Filter  Higher object tracking accuracy and can tolerate slower FPS throughput. 

- [YOLO](https://arxiv.org/abs/1506.02640)  object detection model.  

- If the algorithm doesn't perform well then we can train a deep learning model for object detection or we can pick a pre-trained model and fine-tune it on our generated Blender image data. 




## Important concepts for object detection from a video:

- *Frame Differencing:* Tracking the pixels and the coordinates of the moving object from each frame.

- *Image Thresholding:* Preprocessing images to get better object representation. i.e: Greyscale image.

- *Finding Contours:* Contours are used to identify the shape of an area in the image having the same color or intensity.
  
- *Image Dilation:* Convolution operation on an image wherein a kernel is passed over the entire image to make features more visible. 


## Steps to follow :

1. Apply frame differencing on every pair of consecutive frames.
2. Apply image thresholding on the output image of the previous step.
3. Perform image dilation on the output image of the previous step.
4. Find contours in the output image.
5. Shortlist contours appearing in the detection zone.
6. Save frames along with the final contours.

**Comment by raphaelschwinger** [comment]
@rxxxxxxb why do you suggest a CSRT Tracker and not e.g. KFC Tracker? I would suggest that we try out these two first.

**Comment by raphaelschwinger** [comment]
@rxxxxxxb 
I added a new PR for your code https://github.com/raphaelschwinger/cone_reconstruction/pull/24.

CSRT seems to perform better then KCF in my tests as well!
